crates/ruff_linter/src/rules/pyupgrade/rules/outdated_version_block.rs

```diff
@@ -353,31 +357,29 @@ fn fix_always_true_branch(
     }
 }
 
-/// Converts a `BigInt` to a `u32`. If the number is negative, it will return 0.
-fn bigint_to_u32(number: &BigInt) -> u32 {
-    let the_number = number.to_u32_digits();
-    match the_number.0 {
-        Sign::Minus | Sign::NoSign => 0,
-        Sign::Plus => *the_number.1.first().unwrap(),
-    }
-}
-
 /// Gets the version from the tuple
-fn extract_version(elts: &[Expr]) -> Vec<u32> {
-    let mut version: Vec<u32> = vec![];
+fn extract_version(elts: &[Expr]) -> Option<Vec<isize>> {
+    let mut version: Vec<isize> = vec![];
     for elt in elts {
-        if let Expr::Constant(ast::ExprConstant {
-            value: Constant::Int(item),
+        let Expr::Constant(ast::ExprConstant {
+            value: Constant::Int(int),
             ..
         }) = &elt
-        {
-            let number = bigint_to_u32(item);
-            version.push(number);
-        } else {
-            return version;
+        else {
+            return None;
+        };
+
+        match int {
+            Int::Small(int) => {
+                version.push(*int);
+            }
+            Int::Big(int) => {
+                warn!("Unsupported version number: {}", int);
```

Comment:
I would remove this warning message as it is not helpful to users. They do not indicate where the unsupported version number appears in their source. It ends up being one of those warnings that you can't get rid of and end up wasting a ton of time on.